crates/sc-consensus-subspace-rpc/src/lib.rs

```diff
@@ -95,6 +97,9 @@ pub trait SubspaceRpcApi {
 
     #[method(name = "subspace_acknowledgeArchivedSegment")]
     async fn acknowledge_archived_segment(&self, segment_index: u64) -> RpcResult<()>;
+
+    #[method(name = "subspace_recordsRoot")]
+    async fn records_root(&self, segment_index: u64) -> RpcResult<Option<Sha256Hash>>;
```

Comment:
I think this should take a vector of segment indexes and return a vector of roots to improve efficiency, otherwise there might be too many requests necessary.

Reply:
Yes, sure. However, it seems vulnerable to a large indexes vector. It makes sense to introduce `MAX_SEGMENT_INDEXES_PER_REQUEST` const to limit both farmer and node sides.

Reply:
Certainly. Requester needs to limit number of roots requested or else response will not fit into max size configured in RPC server and response will fail. For public RPC endpoints we also may want to limit this, so yeah, makes sense to have certain limits on both sides.